includes/class-omise-money.php

```diff
@@ -0,0 +1,87 @@
+<?php
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * @since 3.5
+ */
+class Omise_Money {
+	/**
+	 * @var float
+	 */
+	protected $amount;
+
+	/**
+	 * @var string
+	 */
+	protected $currency;
+
+	/**
+	 * @var array
+	 */
+	private $subunit_multiplier = array(
+		'AUD' => 100,
+		'CAD' => 100,
+		'CHF' => 100,
+		'CNY' => 100,
+		'DKK' => 100,
+		'EUR' => 100,
+		'GBP' => 100,
+		'HKD' => 100,
+		'JPY' => 1,
+		'MYR' => 100,
+		'SGD' => 100,
+		'THB' => 100,
+		'USD' => 100
+	);
+
+	/**
+	 * @param int|float|string $amount
+	 * @param string           $currency
+	 */
+	public function __construct( $amount, $currency ) {
+		if ( ! isset( $this->subunit_multiplier[ strtoupper( $currency ) ] ) ) {
+			throw new Exception( __( 'We do not support the currency you are using.', 'omise' ) );
+		}
+
+		$this->amount   = $this->purify_amount( $amount );
+		$this->currency = strtoupper( $currency );
+	}
+
+	/**
+	 * @param  int|float|string $amount
+	 *
+	 * @return float
+	 */
+	public function purify_amount( $amount ) {
+		if ( ! is_string( $amount ) && ! is_float( $amount ) && ! is_numeric( $amount ) ) {
+			throw new Exception( __( 'An amount has to be integer, float, or string.', 'omise' ) );
+		} 
+
+		if ( is_string( $amount ) ) {
+			$amount = preg_replace("/[^0-9.]/", '', $amount);
+		}
+
+		return (float) $amount;
+	}
+
+	/**
+	 * @return int
+	 */
+	public function to_subunit() {
```

Comment:
No subunit to amount function?

Reply:
One example of where this is needed is showing the user limitations of instalment payments (from capabilities) - which are expressed in subunits

Reply:
> No subunit to amount function? 
there is `getAmount()` which returns a original amount that is passed through this class.

> One example of where this is needed is showing the user limitations of instalment payments (from capabilities) - which are expressed in subunits
Sorry, can you explain a bit more? Not sure if I get it.

Reply:
If a limit for a certain payment type is given to us in subunits, we may obviously need to present this to the user as a readable amount. For this, we obviously need to be able to convert from subunits to amount

Reply:
ic. Didn't think of the case converting another way around (Omise API to frontend).
I feel a bit off if "converting Omise API to readable format" method have to be implemented here (as it wasn't designed for).

Let me think a bit. 
In the meantime, any suggestion? (I'm trying to avoid such as `Omise_Utility`, `Omise_Helper` classes)